pkgs/tools/text/sd/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, fetchFromGitHub, rustPlatform, cmake
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "sd";
+  version = "0.5.0";
+
+  src = fetchFromGitHub {
+    owner =  "chmln";
+    repo =  pname;
+    rev = version;
+    sha256 =  "1y44qizciir75d1srwm1mlskhflab2b6153d19vblw410in82f5d";
+  };
+
+  cargoSha256 = "1gls68lw8a2c3gsav70l2wasrgav68q5w1nf50jsrbqq9kb4i7nb";
```

Comment:
I get `0lg8lm4rv9ai61621l6pm19hnj7xzbiaj9v46ihfij1bw7ii54v4` instead.

Reply:
is `0lg8lm4rv9ai61621l6pm19hnj7xzbiaj9v46ihfij1bw7ii54v4` kernel-specific or something? I'm currently building this on darwin, and changing either the `sha256` or the `cargoSha256` to `01g...` results in a sha mismatch and failed build.  
If this is supposed to be the cargoSha256, is there a way I can check this via command line (like with `nix-prefetch-github`?

Reply:
Does it build for you now?

Yeah, I'm talking about `cargoSha256`.
I don't know what the reason could be, I'm building on NixOS.
I don't think there's a command to check the hash.

@Mic92 Do you know what `cargoSha256` depends on? Can we build the same Rust derivations on Linux and MacOS?

Reply:
Yeah, it builds for me. `./result/bin/sd` works fine too. I assume it fails on NixOS?  
I can keep this open and test on a linux machine this weekend.

Reply:
Your new comit works, but only with my hash.
We could use `if stdenv.isDarwin then "yourHash" else "myHash"` as a workaround.
But we should ask someone who knows about the nixpkgs rust infrastructure whether that's normal.

Reply:
This definitely shouldn't happen. The hash is computed from the crate name, version, features, and the (transitive) dependencies:

https://github.com/NixOS/nixpkgs/blob/358c62625a4ea6d8286a2eb15ea03c52ae9cead3/pkgs/build-support/rust/build-rust-crate/default.nix#L116

`Cargo.lock` should fix all dependencies. I think it is more useful to find the source of this difference. Did you test with Nix sandboxing on macOS (if possible)?

Reply:
BTW, builds fine on my NixOS machine with the cargoSha256 in the PR (`1gls68lw8a2c3gsav70l2wasrgav68q5w1nf50jsrbqq9kb4i7nb`).

Reply:
Hmm, yeah it does work now. I wonder what I had changed.

Reply:
Yup, double-checked on darwin and (arch) linux, both work fine.